Added an observability metric event to the discovery category.

hmadison opened the pull request:
#### Related Issue:

Resolves #1421 / Pairing with @pladamgregory.

#### Description of changes:

This adds an event type which is captures metrics which were collected from a target entity. The goal of this class is capture relevant metrics, such as throughput metrics from netflow collectors, failed login attempt counts, or other "time series" style data which can be used to power heuristic based detections or otherwise aid in investigations.

### Delete once you have confirmed the following: 
1. Did you add a single line summary of changes to `Unreleased` section in the [CHANGELOG.md](https://github.com/ocsf/ocsf-schema/blob/main/CHANGELOG.md) file?
2. Have you followed the [contribution guidelines](https://github.com/ocsf/ocsf-schema/blob/main/CONTRIBUTING.md)?
3. Did you run a local instance of the [ocsf-server](https://github.com/ocsf/ocsf-server) and ensure it ran without any errors/warnings?
4. Is your PR title in sync with the description?

hmadison committed:
Added an observability metric event to the discovery category.

hmadison commented on dictionary.json:
```diff
@@ -5726,6 +5726,11 @@
     },
+    "target_entity": {
+      "caption": "Target Entity",
+      "description": "The entity which produced the event.",
```
`target_entity`, and it's smaller descendant `observed_entity` can be though of as a superset of `metadata.loggers`. The idea is that when metric data is generated, there are a set of labels associated with it that can be mapped into common OCSF entities. In this way, we can re-use the same objects which cybersecurity analysts are familiar with to also capture the observability data which is present from a device.

hmadison committed:
Update name to clarify that the event collects a set of metrics.

hmadison committed:
See if "object" behaves better than "_entity".

hmadison committed:
Re-add observed_entity after rebase.

mikeradka commented:
@hmadison Is this PR worth pursuing at this time?

hmadison committed:
Add a missing comma.

hmadison committed:
Extend class from entity

pagbabian-splunk commented on events/discovery/observability_metrics_report.json:
```diff
@@ -0,0 +1,19 @@
+  "name": "observability_metric_report",
+  "attributes": {
+    "observed_entity": {
+      "description": "The entity which produced the metric(s) were collected from. This collection process can happen either via a push from the entity to a third party service or a third party service performing an activity to interface with the entity and extract the metrics of interest.",
```
Small grammar improvement: "The entity which produced the collected metric(s)."

pagbabian-splunk reviewed (dismissed):
See minor description suggestion.

hmadison committed:
Update description per PR feedback.

mikeradka committed:
Merge branch 'main' into hm-ap/observability-metric

Signed-off-by: Mike Radka (Splunk) <91983279+mikeradka@users.noreply.github.com>

mikeradka committed:
Update CHANGELOG.md

Move the line '1. Added new `observability_metric_report` event to the Discovery category. [#1442](https://github.com/ocsf/ocsf-schema/pull/1442)' to 1.8.0 (was in 1.6.0 due to the age of this PR)

Signed-off-by: Mike Radka (Splunk) <91983279+mikeradka@users.noreply.github.com>
